packages/react/src/components/Authenticator/FederatedSignIn/FederatedSignIn.tsx

```diff
@@ -31,7 +34,7 @@ export function FederatedSignIn(): JSX.Element {
                 icon="amazon"
                 key={provider}
                 provider={FederatedIdentityProviders.Amazon}
-                text={translate<string>(`Sign ${federatedText} with Amazon`)}
```

Comment:
Now that we have your lovely `getSignInWithFederationText`, I'm wondering if there's a watch to avoid writing a `switch/case` altogether. Might need a util function to capitalize the provider, but otherwise should be pretty straightforward

Reply:
Right, it needs a map from `SocialProvider` (e.g. 'amazon') to `FederatedIdentityProvider` (e.g. `LoginWithAmazon`). Sounds straightforward, but let's do that in a separate PR to focus on just the text changes on this PR.